### ISSUE TITLE
Linux Support

### ISSUE THREAD

**Issue by frenchiveruti** [JGRennison/OpenTTD-patches#13]
Hello, as far as I understand, there's only a windows compatible patch?
Is there any way to understand how to install this on a Linux OS PC? (more specifically, a Lubuntu PC).
Thanks, and sorry if this is an annoyance.

**Comment by SGunner2014** [comment]
You can compile it on Linux by downloading the source and then running
'./configure' and then 'make'. Both of these need root permissions, so use
sudo.

**Comment by frenchiveruti** [comment]
Hello thanks for the reply, just to be clear I'm not very savvy on the "compiling", I'm surely going to blindly follow what you said.
Just to make sure.
1. Do I download the folder "src" or the entire git?
2. Do I need any extra packets on my linux distro? Or they are around by default.
3. Where do I "compile" I mean, where do I place the "compiled" files.

**Comment by frenchiveruti** [comment]
okokok, It was just a easy as downloading the git and placing the terminal on top of that.
The only problem is the stupid old crap that I have as hardware, that it can't recognize the drivers, and the lastest ones on ShI(t)ntel are from 2010.
This is the error I get on "compile":  

> checking awk... awk  
> detecting OS... UNIX  
> checking build system type... i686-linux-gnu  
> checking host system type... i686-linux-gnu  
> checking universal build... no  
> checking build cc... gcc  
> checking host cc... gcc  
> checking build c++... g++  
> checking host c++... g++  
> checking strip... disabled  
> checking builtin depend... yes  
> checking makedepend... disabled  
> detecting cpu-type... 32 bits  
> detecting SSE... found  
> checking static... no  
> checking unicode... no  
> using debug level... no  
> using desync debug level... no  
> using link time optimization... no  
> checking OSX sysroot... not OSX, skipping  
> checking allegro... not found  
> checking sdl... not found  
> checking COCOA... not OSX, skipping  
> checking GDI video driver... not Windows, skipping  
> configure: error: no video driver development files found

**Comment by auge8472** [comment]
> So yeah, can't use the patch, thanks for your help!

I play the patchpack on two devices with Ubuntu as OS. So it's definitely possible to use the patch(pack).

Read the [compile on debian/Ubuntu](https://wiki.openttd.org/Compiling_on_(GNU/)Linux_and_*BSD#Debian.2C_Ubuntu) section in the OpenTTD-manual. Execute the complete command line instruction (`sudo apt-get install ...`) that is shown in the subsection *"Manually"* with **one exception**. Replace "subversion" with "git" The instruction will install *all* needed packages.

Create a directory, i.e. "jgrpp" in your home directory, enter it and [clone the repository](https://help.github.com/articles/cloning-a-repository/) (only step 6 of the Github-manual with the URL of *this* repository). enter the directory of the cloned repo and run, as also described [in the OpenTTD-manual](https://wiki.openttd.org/Compiling_on_(GNU/)Linux_and_*BSD#3._Compiling_commands.2C_running_the_compiled_binary), again `./configure` and `make` afterwards. When the `make` run is finished, you'll find the binary in the subfolder `bin`.

If you have installed a OpenTTD-version from a Debian-package before, the self compiled patchpack will use the directories in the user space (for GRF, screenshots, savegames and so on) from the old installation.

**Comment by frenchiveruti** [comment]
Hi! I'll try this, I'm pretty sure it'll fail horribly again, because it's not an issue with the Ubuntu OS (Lubuntu in this case), but with my integrated _"graphics"_ card, as it has a veeery old and terribly made driver that it's even worse on linux.
I'll try to make it run later today, but I think the issue is on another place.
Thanks for the time to make a nice reply.

**Comment by SGunner2014** [comment]
The issue here is that you do not have the SDL development files installed

**Comment by frenchiveruti** [comment]
Ah.... I'll try again then, for future reference is the package installed via "apt-get install libsdl2-dev".
It didn't work, it still states the same, I'm going to try the @auge8472 solution to see if that works.

**Comment by frenchiveruti** [comment]
Hello again!
The patch was successful! I just had to run the command:
`sudo apt-get install build-essential pkg-config libsdl1.2-dev subversion patch zlib1g-dev liblzo2-dev liblzma-dev libfontconfig-dev libicu-dev`
That installed the necessary packages.
Then I ran './configure' and 'make' and it worked like a charm.
Thanks for the help both of you and sorry for the inconveniece, it wasn't a problem with JRG patch, but a dumbness one!
